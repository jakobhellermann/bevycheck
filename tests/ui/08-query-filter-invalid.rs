use bevy::prelude::*;

fn main() {
    some_system.system();
}

struct Player;
struct SpawnPlayer;
struct Test<T>(T);
#[bevycheck::system]
fn some_system(query: Query<Entity, Or<(Player, SpawnPlayer)>>) {}
