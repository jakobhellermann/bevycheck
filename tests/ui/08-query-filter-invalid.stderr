error: unknown query filter
  --> $DIR/08-query-filter-invalid.rs:11:41
   |
11 | fn some_system(query: Query<Entity, Or<(Player, SpawnPlayer)>>) {}
   |                                         ^^^^^^
   |
   = note: `Player` is not a known query filter
   = help: if you want to check for Player's existence, use `With<Player>``
